rmf_inorbit_fleet_adapter/rmf_inorbit_fleet_adapter/RobotCommandHandle.py

```diff
@@ -465,8 +467,17 @@ def get_remaining_waypoints(self, waypoints: list):
         may be modified if waypoints in a path need to be filtered.
         '''
         assert (len(waypoints) > 0)
-        remaining_waypoints = []
 
+        # Occasionally, RMF will send two consecutive waypoints with the same values in the `waypoints` argument
+        # of RobotCommandHandle::follow_new_path(), causing an excess of navigation requests from the fleet adapter
+        # The following lines will filter them out
```

Comment:
Is there any related RMF issue we could link here?

Reply:
Yes, here https://github.com/open-rmf/rmf_ros2/issues/272

I am not sure I would link it in the code, I would prefer to link this issue instead. But LMK what you think is best and I'll ad it

Reply:
Linking this issue sounds good!

Reply:
I was given a good explanation for the reasoning behind the duplicated waypoints here https://github.com/open-rmf/rmf_ros2/issues/272#issuecomment-1548925770. I sitll think we should filter them out, but the reason is now documented in the code. I don't think liking this issue is necessary anymore.
I also updated the issue description with the correct information